crates/net/network/src/transactions/mod.rs

```diff
@@ -1082,84 +1082,113 @@ where
     fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
         let this = self.get_mut();
 
-        // drain network/peer related events
-        while let Poll::Ready(Some(event)) = this.network_events.poll_next_unpin(cx) {
-            this.on_network_event(event);
-        }
+        let mut budget = 1024;
 
-        // drain commands
-        while let Poll::Ready(Some(cmd)) = this.command_rx.poll_next_unpin(cx) {
-            this.on_command(cmd);
-        }
+        loop {
+            let mut some_ready = false;
```

Comment:
I'd like to rename this:
```suggestion
            let mut maybe_more_work = false;
```

Reply:
I was thinking about the existential quantifier